app/controllers/api/v0/versions_controller.rb

```diff
@@ -25,6 +25,11 @@ def show
     }
   end
 
+  def raw
+    @version ||= version_collection.find(params[:id])
+    redirect_to @version.uri, status: 301 and return
```

Comment:
I like that turning the `Version#uri` field into a public URL keeps things simple, but given the above issue about what/how different storage implementations might do things, I’m unsure whether that’s really a feasible approach.

- My first gut feeling is that we should be conceiving of `Version#uri` as a URI that we can just pass to a `FileStorage` implementation (maybe it even has a scheme that identifies which implementation?), and different implementations will parse it in different ways to go get what they need. When we serialize a `Version` (i.e. in `Version#as_json`), we should be replacing `uri` with `/api/v0/versions/{uuid}/raw` or `/api/v0/raw/{version_hash}`. Then the route handler at that path figures out what to do by passing the *actual* `uri` from the database to the FileStorage.

- It’s possible we could still store a public URL here, but I feel like it would still need to incorporate FileStore-specific info, since there’s nowhere else to put it. i.e. we could have URLs like `/api/v0/raw/{some-identifier-or-path-or-url-fragment-provided-by-and-passed-back-to-a-filestorage-implementation}`.

    I think the reason I don’t immediately jump to this approach is because it feels like we are freezing part of our API/URL structure into the database, which feels like a coupling that is way too tight. The API is at `v0` for a reason ;)